Feature: Lists separated by solvency

tokdaniel opened the pull request:
- Added solvency tier tags to each token in the full list, and a script to split and validate based on this (and adding underlying tokens)
- Removed bridgeInfo from L2-s. Mainnet bridgeInfo-s are kept

kasparkallas commented on scripts/split-by-solvency.ts:
```diff
@@ -0,0 +1,71 @@
+    if (
+      ["Wrapper", "Native Asset"].includes(
+        token.extensions?.superTokenInfo?.type ?? ""
+      )
```
Approaching from `token.extensions?.superTokenInfo?.underlyingTokenAddress` is slightly easier in the sense that you don't even have to know the type of the token. If it has an underlying token then include it in the list.

tokdaniel replied:
that's not a common property between the types.

kasparkallas commented on scripts/split-by-solvency.ts:
```diff
@@ -0,0 +1,71 @@
+          // @ts-ignore (Special Native Assets have underlying tokens)
+          token.extensions?.superTokenInfo?.underlyingTokenAddress
+      );
+      if (underlyingToken) {
```
I'm assuming the validation for all Super Tokens to have an underlying token will happen later.

tokdaniel replied:
yes, the validation happens after splitting.

kasparkallas commented on scripts/split-by-solvency.ts:
```diff
@@ -0,0 +1,71 @@
+          token.extensions?.superTokenInfo?.underlyingTokenAddress
+      );
+      if (underlyingToken) {
+        tokens.push(underlyingToken);
```
Technically there is some small chance for duplicate underlying tokens getting added. It might be validated later in the process though.

tokdaniel replied:
modified the find, to also include chainId

kasparkallas commented on scripts/split-by-solvency.ts:
```diff
@@ -0,0 +1,71 @@
+  });
+
+  try {
+    const partialTokenList = { ...tokenList, tokens };
```
The reason to spread the initial `tokenList` is to get the metadata from there?

tokdaniel replied:
yes

kasparkallas commented on scripts/split-by-solvency.ts:
```diff
@@ -0,0 +1,71 @@
+  fs.writeFileSync(
+    `solvency-${tier}.tokenlist.json`,
+    JSON.stringify(
+      {
```
It's a bit weird code structure to put together a token list, validate it, and then put it together again. I'd prefer the code to flow in a way it's put together once.

tokdaniel replied:
just an overlook, i was in the middle of refactoring

kasparkallas commented:
Currently, the file names are:
```
superfluid.tokenlist.json
solvency-tier_a.tokenlist.json
solvency-tier_b.tokenlist.json
solvency-tier_c.tokenlist.json
```

It's nitpicky but I wish they looked slightly better and more aligned. Something like:
```
superfluid.tokenlist.json
superfluid.tier-a.tokenlist.json
superfluid.tier-b.tokenlist.json
superfluid.tier-c.tokenlist.json
```

kasparkallas commented:
I'm thinking let's just remove the `bridgeInfo` from Super Tokens completely.

This utils package describes:
"This package contains utility functions used for converting an L1 token list into a cross-chain tokenlist by adding cross-chain mapping information for Arbitrum, Optimism, and Polygon to a given L1 list."
Source: https://github.com/Uniswap/token-list-bridge-utils

Because we're not converting any L1 list then I'd say no need for it after all. The `bridgeInfo` makes more sense for the underlying tokens. Or for Pure Super Tokens at best.

tokdaniel commented:
> Currently, the file names are:
> 
> ```
> superfluid.tokenlist.json
> solvency-tier_a.tokenlist.json
> solvency-tier_b.tokenlist.json
> solvency-tier_c.tokenlist.json
> ```
> 
> It's very nitpicky but I wish they looked slightly better and more aligned. Something like:
> 
> ```
> superfluid.tokenlist.json
> superfluid.tier-a.tokenlist.json
> superfluid.tier-b.tokenlist.json
> superfluid.tier-c.tokenlist.json
> ```

i'd keep the _, so `superfluid.tier_a.tokenlist.json`, but other than that sure. The reason is, that `_` is allowed in tags, while `-` is not, and It would probably be nice, to keep the namings the same every in every location.

vmichalik commented:
If possible I'd like to see the standard superfluid.tokenlist.json include
tiers A and B and not C. that way the default pick for integrations will
always exclude low quality tokens

On Tue, Sep 26, 2023 at 1:11 PM tokdaniel ***@***.***> wrote:

> Currently, the file names are:
>
> superfluid.tokenlist.json
> solvency-tier_a.tokenlist.json
> solvency-tier_b.tokenlist.json
> solvency-tier_c.tokenlist.json
>
> It's very nitpicky but I wish they looked slightly better and more
> aligned. Something like:
>
> superfluid.tokenlist.json
> superfluid.tier-a.tokenlist.json
> superfluid.tier-b.tokenlist.json
> superfluid.tier-c.tokenlist.json
>
> i'd keep the _, so superfluid.tier_a.tokenlist.json, but other than that
> sure. The reason is, that _ is allowed in tags, while - is not, and It
> would probably be nice, to keep the namings the same every in every
> location.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/superfluid-finance/tokenlist/pull/30#issuecomment-1735956229>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC7B54LQHIDLY3FDG2YZ6FLX4MEDRANCNFSM6AAAAAA5HN7XDQ>
> .
> You are receiving this because your review was requested.Message ID:
> ***@***.***>
>


-- 
*Vijay Michalik*
Head of Product at Superfluid

*--*

*This communication has been sent from Superfluid Finance Ltd. Its contents
and any files attached are confidential and may contain privileged
information. If you are not the intended recipient please notify the sender
immediately by reply e-mail and please delete the message from your system
immediately. You should not copy it or use it for any purpose nor disclose
its contents to any other person. Any unauthorized use of the content of
this message is a breach of your duty to respect the confidentiality of the
correspondence between other persons and can expose the responsible party
to civil and/or criminal penalties, and may constitute a more serious
offense.*

tokdaniel commented:
> If possible I'd like to see the standard superfluid.tokenlist.json include tiers A and B and not C. that way the default pick for integrations will always exclude low quality tokens
> […](#)
> On Tue, Sep 26, 2023 at 1:11 PM tokdaniel ***@***.***> wrote: Currently, the file names are: superfluid.tokenlist.json solvency-tier_a.tokenlist.json solvency-tier_b.tokenlist.json solvency-tier_c.tokenlist.json It's very nitpicky but I wish they looked slightly better and more aligned. Something like: superfluid.tokenlist.json superfluid.tier-a.tokenlist.json superfluid.tier-b.tokenlist.json superfluid.tier-c.tokenlist.json i'd keep the _, so superfluid.tier_a.tokenlist.json, but other than that sure. The reason is, that _ is allowed in tags, while - is not, and It would probably be nice, to keep the namings the same every in every location. — Reply to this email directly, view it on GitHub <[#30 (comment)](https://github.com/superfluid-finance/tokenlist/pull/30#issuecomment-1735956229)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AC7B54LQHIDLY3FDG2YZ6FLX4MEDRANCNFSM6AAAAAA5HN7XDQ> . You are receiving this because your review was requested.Message ID: ***@***.***>
> -- *Vijay Michalik* Head of Product at Superfluid *--* *This communication has been sent from Superfluid Finance Ltd. Its contents and any files attached are confidential and may contain privileged information. If you are not the intended recipient please notify the sender immediately by reply e-mail and please delete the message from your system immediately. You should not copy it or use it for any purpose nor disclose its contents to any other person. Any unauthorized use of the content of this message is a breach of your duty to respect the confidentiality of the correspondence between other persons and can expose the responsible party to civil and/or criminal penalties, and may constitute a more serious offense.*

changed, and added a superfluid.extended.tokenlist.json
